### ISSUE TITLE
unified-signatures suggests combining declarations that shouldn't be

### ISSUE THREAD

**Issue by zbjornson** [Microsoft/dtslint#183]
```ts
class MyClass {
  method1(mimeType: 'image/png', config?: PngConfig): void
  method1(mimeType: 'image/jpeg', config?: JpegConfig): void

  /** call with a number */
  method2(p1: number): void
  /** call with a string */
  method2(p1: string): void
}
```

dtslint suggests combining both of those pairs of declarations, but I think that is incorrect. `method1` shouldn't be combined because the type of the 2nd arg depends on the value of the 1st arg. `method2` shouldn't be combined because their doc strings are different.

**Comment by ndunks** [comment]
There's no explanation how to face this? I also have same experience..
